### PR TITLE
test(components): utility modules and behavioral directives coverage (#1679)

### LINKED ISSUE #1679: Coverage: components Utility Modules and Behavioral Directives
## Parent

GSA/sam-design-system#1640

## What to build

Expand test coverage for utility services and behavioral directives in `components` (click-outside, tab-outside, key-helper, autoclose, positioning, and triggers). Exercise event listeners, outside-click detection, and positioning calculations.

## Acceptance criteria

- [ ] Spec(s) added/extended for utility directives and helper functions
- [ ] Branch and function coverage improved for `components`
- [ ] Tests pass cleanly with `npm run test:components` and `npm run coverage:check`

## Blocked by

None - can start immediately

### PR DESCRIPTION
**Pull request by fpigeonjr**
## Description

Expanded unit test coverage for utility modules and behavioral directives in `components`:
- Covered `click-outside`, `tab-outside`, and `key-helper` across key codes, number parsing, and edge cases.
- Created test suite for `observe-width.directive.spec.ts` testing ViewportRuler resize events.
- Created test suite for `autoclose.spec.ts` covering inside/outside clicks, Escape handling, and ignore selectors.
- Created test suite for `triggers.spec.ts` covering hover/focus and custom trigger delays.
- Created test suites for `positioning`, `rtl`, `util`, `ngbTransition`, and `popup`. Targeted modules achieved ~100% lines. 485 tests passing.

## Motivation and Context

Closes #1679
Part of EPIC #1640: Raise sam-design-system test coverage to meet and exceed QASP quality gates.

## Type of Change (Select One and Apply Github Label)

- [ ] Bug fix (non-breaking change which fixes an issue) -> Apply bugfix label
- [x] New feature (non-breaking change which adds functionality) -> Apply enhancement label
- [ ] Breaking change (fix or feature that would cause existing functionality to change) -> Apply breaking label

## Screenshots (if appropriate):

N/A — unit test additions only.

## Which browsers have you tested?

- [x] Automated tests run in jsdom / Vitest

## Checklist:

- [x] I have read the [CONTRIBUTING.md](https://github.com/GSA/sam-design-system/blob/master/CONTRIBUTING.md) document.
- [x] My code passes the automated a11y lint gate.
- [ ] This code has been reviewed by another team member and passes the reviewer checklist found in [CONTRIBUTING.md](https://github.com/GSA/sam-design-system/blob/master/CONTRIBUTING.md).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests pass; the coverage floor has not regressed.
- [x] Lint baseline not raised.
- [x] My code conforms to WCAG 2.1 AA.
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/packages/components/src/lib/observers/observe-width.directive.spec.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import { Component } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { ViewportRuler } from '@angular/cdk/overlay';
+import { Subject } from 'rxjs';
+
+import { SdsObserveWidthDirective } from './observe-width.directive';
+
+@Component({
+  template: ` <div [sdsObserveWidth] (sdsObserveWidth)="onWidthChange($event)">Target Element</div> `,
```
[the remark this answers is not in the document]
Comment (fpigeonjr) [reply libs/packages/components/src/lib/observers/observe-width.directive.spec.ts]: Fixed in 934344c61: applied sdsObserveWidth as a bare attribute directive while preserving event binding

File: libs/packages/components/src/lib/util/autoclose.spec.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  let closed$: Subject<void>;
+  let closeSpy: any;
+  let container: HTMLElement;
+  let insideEl: HTMLElement;
+  let outsideEl: HTMLElement;
+  let ignoreEl: HTMLElement;
+
+  beforeEach(() => {
+    zone = TestBed.inject(NgZone);
+    closed$ = new Subject<void>();
```
[the remark this answers is not in the document]
Comment (fpigeonjr) [reply libs/packages/components/src/lib/util/autoclose.spec.ts]: Fixed in 934344c61: added TestBed.configureTestingModule({}) before injecting NgZone

File: libs/packages/components/src/lib/util/util.spec.ts [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+        const shadow = host.attachShadow({ mode: 'open' });
+        const shadowBtn = document.createElement('button');
+        shadow.appendChild(shadowBtn);
+
+        // Mock shadow active element
+        Object.defineProperty(host, 'shadowRoot', { value: shadow, configurable: true });
+        Object.defineProperty(shadow, 'activeElement', { value: shadowBtn, configurable: true });
+        Object.defineProperty(document, 'activeElement', { value: host, configurable: true });
+
+        expect(getActiveElement(document)).toBe(shadowBtn);
```
[the remark this answers is not in the document]
Comment (fpigeonjr) [reply libs/packages/components/src/lib/util/util.spec.ts]: Fixed in 934344c61: used a local stub root with activeElement: host without mutating document.activeElement
